### PR TITLE
Use TryAdd to add services rather than Add

### PR DESCRIPTION
**Pull request by kichalla**
@HaoK

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@kichalla__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Kiran Challa). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: src/Microsoft.AspNet.Cors.Core/CorsServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -35,8 +35,10 @@ public static class CorsServiceCollectionExtensions
         public static IServiceCollection AddCors(this IServiceCollection serviceCollection)
         {
             serviceCollection.AddOptions();
-            serviceCollection.AddTransient<ICorsService, CorsService>();
-            serviceCollection.AddTransient<ICorsPolicyProvider, DefaultCorsPolicyProvider>();
+            serviceCollection.TryAdd(
+                ServiceDescriptor.Transient(typeof(ICorsService), typeof(CorsService)));
```
Comment (HaoK) [inline src/Microsoft.AspNet.Cors.Core/CorsServiceCollectionExtensions.cs]: Can you not use the generic overload of transient?

Comment (kichalla) [inline src/Microsoft.AspNet.Cors.Core/CorsServiceCollectionExtensions.cs]: Sure, will change it...

**Comment by kichalla** [comment]
Updated. /cc: @HaoK

**Comment by HaoK** [comment]
:shipit:

**Comment by kichalla** [comment]
Commit: 1cadc54c3b3fca71fcabd90965d2dd76ad1ebaf0

**Pull request closed without merging**